src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -489,8 +501,13 @@ private void handleCloudWatchAttemptStatus(CloudWatchAttempt cloudWatchAttempt)
                 return;
             }
 
+            ProcessingFiles processingFiles = processingFilesInformation.get(componentName);
+
             // Delete files based on diskspace
-            this.diskSpaceManagementService.freeDiskSpace(attemptLogInformation.getLogFileGroup());
+            List<String> deletedHashes =
+                    this.diskSpaceManagementService.freeDiskSpace(attemptLogInformation.getLogFileGroup());
+
+            processingFiles.remove(deletedHashes); // Stop tracking files already uploaded
```

Comment:
Remove entries from the cache for files that have been removed due to disk space being freed